src/cdk/stepper/stepper.ts

```diff
@@ -210,27 +215,26 @@ export class CdkStepper {
   }
 
   private _emitStepperSelectionEvent(newIndex: number): void {
-    const stepsArray = this._steps.toArray();
     this.selectionChange.emit({
       selectedIndex: newIndex,
       previouslySelectedIndex: this._selectedIndex,
-      selectedStep: stepsArray[newIndex],
-      previouslySelectedStep: stepsArray[this._selectedIndex],
+      selectedStep: this._stepsArray[newIndex],
+      previouslySelectedStep: this._stepsArray[this._selectedIndex],
     });
     this._selectedIndex = newIndex;
   }
 
   _onKeydown(event: KeyboardEvent) {
     switch (event.keyCode) {
       case RIGHT_ARROW:
-        if (this._dir && this._dir.value === 'rtl') {
+        if (this._layoutDirection() === 'rtl') {
           this._focusStep((this._focusIndex + this._steps.length - 1) % this._steps.length);
```

Comment:
since these previous and next expressions are repeated and kind of intimidating it might be nice for readability  to create `_focusNextStep` and `_focusPreviousStep` methods